pkg/agent/proxy/proxier.go

```diff
@@ -258,12 +269,40 @@ func (p *proxier) installServices() {
 		}
 
 		var endpointUpdateList []k8sproxy.Endpoint
-		for _, endpoint := range endpoints { // Check if there is any installed Endpoint which is not expected anymore.
-			if _, ok := endpointsInstalled[endpoint.String()]; !ok { // There is an expected Endpoint which is not installed.
-				needUpdateEndpoints = true
+		if len(endpoints) > maxEndpoints {
```

Comment:
personally I would have liked to see a unit test for this (with `maxEndpoints` set to small value). However, this is strictly better than what we had before, and there is little risk in merging this.

Reply:
Make sense. I prefer to add a unit test. However,  the `maxEndpoints` is a constant variable, IMO, I didn't come up with any good idea to test this. BTW, thanks for merging this.